### PR TITLE
Msbuild target to copy translation files.

### LINKED ISSUE #4329: Localization packages need some msbuild scripts
About the localization packages.

As i remember, with the new nuget package system content files are not copied under the project, they are immutable contents that may be referenced but as link items. Hmm, maybe with the right option they can be outputed on building but they will be under the bin folder, not under the project root.

So, i think that the localization packages need to be packed with some msbuild `*.props and / or *.targets` scripts so that `*.po` files are copied on building and then on publishing.

As we did at some point for modules when they needed to be under the application project root.

### PR DESCRIPTION
**Pull request by jtkech**
Fixes #4329 

- So, here i added a target in `OC.Application.Cms.Core.Targets.targets` so that translations files, from referenced translation packages (even transitively), are copied under the project root folder.

- I also updated the place where we mark them to be published so that it works even we do a publish without having done any 1st build before.

- I thought about a target to clean the localization files before building or on a cleaning cmd, but i removed it to let the ability to manualy add some translation files under the `Localization` folder.

**What we have to do in the translations repo**

- No need to change the `*.nuspec` template.

- We just need to create somewhere the following shared `Package.Build.props` file. Then, when creating a translation package, this file need to be copied under the package `buildTransitive` folder and renamed to `{PackageId}.props` => `buildTransitive/{PackageId}.props`.

      <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
        <ItemGroup>
          <PackageTranslationFiles Include="$(MSBuildThisFileDirectory)..\content\Localization\**\*" />
        </ItemGroup>
      </Project>

### CODE REVIEW

File: src/OrchardCore.Build/Dependencies.props [hunk 1]
```diff
@@ -27,7 +27,7 @@
     <PackageManagement Include="Newtonsoft.Json" Version="12.0.2" />
     <PackageManagement Include="NLog.Web.AspNetCore" Version="4.8.5" />
     <PackageManagement Include="NodaTime" Version="2.4.7" />
-    <PackageManagement Include="OrchardCore.Translations.All" Version="1.0.0-beta3-100243" />
+    <PackageManagement Include="OrchardCore.Translations.All" Version="1.0.0-beta3-100246" />
```
Comment (hishamco) [inline src/OrchardCore.Build/Dependencies.props]: 👍

**Comment by jtkech** [comment]
So, after updating the translations package version embedding the above props files, i re-built the solution and run it, then after enableing the loalization feature i could see some french translations.

FrenchTranslation

Setting `Show All Files` under VS i could see

FrenchTranslation1

**Comment by hishamco** [comment]
Woohoo 🚀🚀🚀 

Great job Jean

**Comment by jtkech** [comment]
You too Hisham, you did most of the things in the translations repo ;)

**Comment by hishamco** [comment]
Just one question, does this will work for all the localization cases: tenant, modular .. etc, coz the localization of .po files is different

**Comment by jtkech** [comment]
No, for tenant specific translations the po files will need to be put manually in the right place. Idem for a module specific translations files that will be embedded in the module assembly and then also available in a published context.

These files are app level translations that will be used by any tenant if no other specific files are there.

**Comment by hishamco** [comment]
Thanks for clarification

**Comment by sebastienros** [comment]
I will run a benchmark since now it's been merged. But I think I might remove the dependency on .All for now and let people reference it or any language, only to be certain that we did what we could to prevent all tenants from loading all cultures.

**Comment by jtkech** [comment]
Yes, see [my comment here](https://github.com/OrchardCMS/OrchardCore.Translations/issues/28#issuecomment-533843984)

> After a quick look, seems that all transaltions from all providers are loaded an cached per tenant. Yes, we would need some static cache shared across tenants as we do for other things as shape tables. I think also for the translations provided by a given module that may be enabled in multiple tenants.

> Only tenant specific translations would needs to be cached per tenant.

---

> and let people reference it or any language

i agree

> we did what we could to prevent all tenants from loading all cultures.

I don't think so, but after a quick look, so not sure

**Pull request merged**